src/scripts/modules/list.js

```diff
@@ -27,14 +32,13 @@ function bindEvents() {
   document.addEventListener("DOMContentLoaded", setActive);
   document.addEventListener("scroll", shrinkNav);
   window.addEventListener("scroll", restoreNav);
-  events.subscribe("destinationAdded", (itemIndex) => {
-    const elDestination = renderDestination(destinationsList[itemIndex]);
+  events.subscribe("destinationUpdate", (newItem) => {
+    const elDestination = renderDestination(newItem);
     gallery.appendChild(elDestination);
   });
```

Comment:
A hint: To make things easy when mutating the DOM with the new list (eg a new item or a modified item), we'll change this logic. Instead of receveing just the `newItem`, it will receive the `newList` (that contains the entire list).

Then with that, we update `storedList` with the received `newList` and call `render()` again. As usual, give it a try before checking the solution ;)

<details>
<summary>Solution:</summary>

```js
events.subscribe("destinationUpdate", (newList) => {
  // Update the list with the new list state
  storedList = newList

  // Call render again to refresh the gallery with the fresh storedList
  render()
});
```

</details>